_posts/2023-07-25-embulk-gem-install-in-v0_9-series-sometime-fails.md

```diff
@@ -0,0 +1,65 @@
+---
+layout: posts
+title: "embulk gem install in v0.9 series sometime fails even though the target plugin exists."
+date: 2023-07-25
+last_modified_at: 2023-07-25
+description: embulk gem install in v0.9 series sometime fails even though the target plugin exists. And how to solve this issue.
+author: "hiroyuki-sato"
+---
+
+## Overview
+
+ Rubygems had been removed old APIs(see links below). These APIs where used in the Embulk 0.9.x series. As a result, `embulk gem install` fails even if the specified gem exists.
+ This article writes how to resolve this issue.
+
+## An example error.
+
+ The following output tried to install `embulk-output-mysql` version 0.10.2, It is the [latest release](https://rubygems.org/gems/embulk-output-mysql/versions/0.10.2-java) that plugin.
+
+ Embulk 0.9.25 raise an error `Could not find a valid gem` even the gem exists.
+This is because Embulk 0.9 serises use old `gem` command. And the command is using deprecated APIs.
+
+```
+embulk gem install embulk-output-mysql -v 0.10.2
+
+================================== [ NOTICE ] ==================================
+ Embulk v0.11.0 will be released soon, planned for June 2023.
+
+ This v0.11.0 will contain a lot of incompatible changes from v0.9.
+ Many plugins are expected to stop working with v0.11.0.
+
+ Try v0.10.48 or later, Release Candidate for v0.11, before v0.11.0 is official.
+
+ See: https://www.embulk.org/articles/2023/04/13/embulk-v0.11-is-coming-soon.html
+================================================================================
+
+2023-07-26 21:41:33.966 +0900: Embulk v0.9.25
+
+Gem plugin path is: /Users/user/.embulk/lib/gems
+
+io/console on JRuby shells out to stty for most operations
+ERROR:  Could not find a valid gem 'embulk-output-mysql' (= 0.10.2) in any repository
+ERROR:  Possible alternatives: embulk-output-mysql
+```
+
+## Plan A: Use embulk 0.11.x.
+
+ New stable version embulk 0.11 had been released at 2023-06-13. It is possible to latest `gem` command if you specify new JRuby version.
+It is the recommendation way to solve this issue.
+
+## Plan B: Use bundle command.
+
+ If you want to use Embulk 0.9 series for some reason, you need to use `embulk bundle` instead of `embulk gem` command.
+ First, you create a `Gemfile` like the one below. after that, execute `embulk bundle` command in the same directory.
```

Comment:
```suggestion
If you cannot start using Embulk v0.11 for some reasons, the other option is using Bundler with Embulk v0.9, instead of the `gem` command. Bundler has been used with Embulk v0.9 or earlier for a long time to manage Embulk plugin installations.

To start using Bundler with Embulk v0.9, you'll have to create `Gemfile` like the following.
```